### PR TITLE
feat: add recent sessions(backend), grouping and chat page

### PR DESCRIPTION
**Pull request by howard9199**
recent sessions:
- return the created and joined sessions from backend

grouping:
- add join column in the join page
- add document 'groups' in firestore
- add group information in  the session

chat page:
- add a sample chat page

### CODE REVIEW

File: src/routes/join/+page.server.ts [hunk 1]
```diff
@@ -35,15 +40,50 @@ export const actions = {
...
 
+		// Add participant to group
+		const groupSnapshot = await adminDb
+			.collection('groups')
+			.where('number', '==', parseInt(groupNumber))
+			.where('sessionId', '==', session.id)
+			.limit(1)
+			.get();
+
+		let groupRef;
```
Comment (ryanlinjui) [inline src/routes/join/+page.server.ts:60]: Suggestions: 
- Variable type
- Initial value

Comment (ryanlinjui) [inline src/routes/join/+page.server.ts:60]: Suggestions: 
- Variable type
- Initial value

File: src/routes/join/+page.server.ts [hunk 2]
```diff
@@ -35,15 +40,50 @@ export const actions = {
...
 		await session.ref.update({
 			[`participants.${locals.user.uid}`]: {
 				name: locals.user.name,
-				joinedAt: new Date()
+				joinedAt: new Date(),
+				groupNumber: parseInt(groupNumber)
 			}
 		});
 
+		// Add participant to group
```
Comment (JacobLinCool) [inline src/routes/join/+page.server.ts:52]:
```suggestion
		// Just teating something
```

File: src/routes/join/+page.server.ts (hunk 1 above)
Comment (ryanlinjui) [inline src/routes/join/+page.server.ts:60]:
```suggestion
		let groupRef: FirebaseFirestore.DocumentReference | null = null;
```

Suggestions: 
- Define variable type
- Initial value

**Review by ryanlinjui — CHANGES_REQUESTED** [review]

**Pull request merged**